### PR TITLE
update environment and address deprecations

### PR DESCRIPTION
**Pull request by fneum**
depending on whether all CI tests pass, I still need to remove the commented lifted version bounds.

### CODE REVIEW

File: envs/environment.yaml [hunk 1]
```diff
@@ -2,19 +2,19 @@
 #
 # SPDX-License-Identifier: MIT
 
-name: pypsa-eur
+name: pypsa-eur-nov14
```
Comment (FabianHofmann) [inline envs/environment.yaml:5]: Not sure about this here, or is it not intended? If we want to do this, I'd rather use the pypsa-eur version as a suffix.

File: envs/environment.yaml [hunk 2]
```diff
@@ -2,19 +2,19 @@
...
 channels:
   - conda-forge
   - bioconda
   - http://conda.anaconda.org/gurobi
 dependencies:
-  - python>=3.8
+  - python=3.8 # fix because gurobi has no builds for python 3.9
```
Comment (FabianHofmann) [inline envs/environment.yaml:11]: Don't think this is necessary. I have python 3.9 and gurobi works smoothly (however cplex is not available on python 3.9)

File: scripts/cluster_network.py [hunk 3]
```diff
@@ -140,6 +140,9 @@
 from pypsa.networkclustering import (busmap_by_kmeans, busmap_by_spectral_clustering,
                                      _make_consense, get_clustering_from_busmap)
 
+import warnings
+warnings.filterwarnings(action='ignore', category=UserWarning)
```
Comment (FabianHofmann) [inline scripts/cluster_network.py:144]: Out of curiosity - what warning is suppressed?

**Review by FabianHofmann — CHANGES_REQUESTED** [review]

File: envs/environment.yaml (hunk 1 above)
Comment (fneum) [reply envs/environment.yaml:5]: Ups. Let's keep pypsa-eur.

File: envs/environment.yaml (hunk 2 above)
Comment (fneum) [reply envs/environment.yaml:11]: Interesting. It works. But why if the builds for py39 are not in the conda channel? https://anaconda.org/Gurobi/gurobi/files

File: scripts/cluster_network.py (hunk 3 above)
Comment (fneum) [reply scripts/cluster_network.py:144]: KMeans user warning that dataset contains features that are not used in the clustering.

Comment (fneum) [reply scripts/cluster_network.py:144]: Maybe @martacki can comment on this deprecation warning and whether it's something we need to address?

**Review by FabianHofmann — APPROVED** [review]

**Pull request merged**